src/Polly.Core/Strategy/PredicateBuilder.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using Polly.Strategy;
+
+namespace Polly;
+
+/// <summary>
+/// Defines a builder for creating predicates for <typeparamref name="TResult"/> and <see cref="Exception"/> combinations.
+/// </summary>
+/// <typeparam name="TResult">The type of the result.</typeparam>
+public sealed class PredicateBuilder<TResult>
+{
+    private readonly List<Predicate<Outcome<TResult>>> _predicates = new();
+
+    /// <summary>
+    /// Adds a predicate for handling exceptions of the specified type.
+    /// </summary>
+    /// <typeparam name="TException">The type of the exception to handle.</typeparam>
+    /// <returns>The same instance of the <see cref="PredicateBuilder{TResult}"/> for chaining.</returns>
+    public PredicateBuilder<TResult> Handle<TException>()
+        where TException : Exception
+    {
+        return Handle<TException>(_ => true);
```

Comment:
I wish there was an analyzer to detect these (unless the compiler automatically does it anyway) 😄

```suggestion
        return Handle<TException>(static _ => true);
```